firewood/src/merkle/node.rs

```diff
@@ -490,6 +495,93 @@ pub enum EncodedNodeType {
     },
 }
 
+// TODO: probably can merge with `EncodedNodeType`.
+#[derive(Debug, Deserialize)]
+struct EncodedBranchNode {
+    chd: Vec<(u64, Vec<u8>)>,
+    data: Option<Vec<u8>>,
+    path: Vec<u8>,
+}
+
+// Note that the serializer passed in should always be the same type as T in EncodedNode<T>.
+impl Serialize for EncodedNode<PlainCodec> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let n = match &self.node {
+            EncodedNodeType::Leaf(n) => {
+                let data = Some(n.data.to_vec()).filter(|d| !d.is_empty());
+                let chd: Vec<(u64, Vec<u8>)> = Default::default();
+                let path = from_nibbles(&n.path.encode(true)).collect();
```

Comment:
🤔, we don't need the prefix to match with `MerkleDB`, right?

Reply:
I think we need, but we can revisit this after we remove the extension node with correct path stored.